packages/engine-formula/src/engine/dependency/formula-dependency.ts

```diff
@@ -609,25 +581,36 @@ export class FormulaDependencyGenerator extends Disposable {
              * includeTree: modification range contains formula
              */
             if (
-                (forceCalculate ||
-                    tree.dependencyRange(
-                        this._updateRangeFlattenCache,
-                        this._dirtyUnitSheetNameMap,
-                        this._currentConfigService.getExcludedRange()
-                    ) ||
-                    this._includeTree(tree)) &&
-                !existTree.has(tree)
+                (
+                    forceCalculate ||
+                    tree.dependencySheetName(this._currentConfigService.getDirtyNameMap()) ||
+                    (this._isInDirtyRange(tree, this._currentConfigService.getDirtyRanges()) && !tree.isExcludeRange(this._currentConfigService.getExcludedRange())) ||
+                    this._includeTree(tree)
+                ) && !existTree.has(tree)
```

Comment:
can you mark the time complex of the comments code every part,  and think is there some way to performance good.